gensim/models/doc2vec.py

```diff
@@ -820,15 +820,15 @@ def _get_offsets_and_start_doctags_for_corpusfile(cls, corpus_file, workers):
         offsets = []
         start_doctags = []
 
-        with smart_open(corpus_file) as fin:
+        with smart_open(corpus_file, encoding='utf8') as fin:
             curr_offset_idx = 0
             prev_filepos = 0
 
             for line_no, line in enumerate(fin):
                 if curr_offset_idx == len(approx_offsets):
                     break
 
-                curr_filepos = fin.tell()
+                curr_filepos = prev_filepos + len(any2utf8(line))
```

Comment:
What is the purpose of this utf=>unicode=>utf8 recoding? It looks like the unicode `line` is never used, so it seems like unnecessary work to first decode it from utf8 to unicode, and then encode that unicode into utf8 again.

Or is it to ensure valid utf8 (exception if file not in utf8)?

Reply:
I just found this solution to be working for all pythons (I used `fin.tell()` but it turned out to work in a wrong way (if you do `fin.readline()` and then `fin.tell()` it will not produce offset to the next line, I don't know why). 

I suppose this is not a final version of this function because it doesn't work on Windows currently, but I'm working on that

Reply:
Fixed, thanks!